### PR TITLE
Dashboard Task Component

### PR DESCRIPTION
**Pull request by JohnLDavidson**
Added Task Component to Dashboard, including frontend foundation and dummy data. Will return to this once API's are better established for connecting with actual data

### CODE REVIEW

File: frontend/crd/src/app/dashboard/events/EventService.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    });
+    service = TestBed.inject(EventService);
+    httpMock = TestBed.inject(HttpTestingController);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  // it('events should return list of events', () => {
```
Comment (bml2238) [inline frontend/crd/src/app/dashboard/events/EventService.spec.ts:24]: please put the tests back

File: frontend/crd/src/app/dashboard/events/events.component.spec.ts [hunk 2]
```diff
@@ -1,53 +1,53 @@
...
-    fixture = TestBed.createComponent(EventsComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
-
-  it('should create', () => {
-    expect(component).toBeTruthy();
-  });
-});
+// describe('EventsComponent', () => {
```
Comment (bml2238) [inline frontend/crd/src/app/dashboard/events/events.component.spec.ts:10]: why did you remove this?

File: frontend/crd/src/app/milestones-page/milestones/milestones.component.spec.ts [hunk 3]
```diff
@@ -10,31 +10,31 @@ import { MatCheckboxModule } from "@angular/material/checkbox";
...
-    fixture = TestBed.createComponent(MilestonesComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
-
-  it('should create', () => {
-    expect(component).toBeTruthy();
-  });
-});
+// describe('MilestonesComponent', () => {
```
Comment (bml2238) [inline frontend/crd/src/app/milestones-page/milestones/milestones.component.spec.ts:13]: why are you removing the tests?

File: frontend/crd/src/app/milestones-page/milestones/milestones.component.ts [hunk 4]
```diff
@@ -28,10 +32,29 @@ export class MilestonesComponent implements OnInit, OnDestroy {
...
+    dialogConfig.height = "80%";
+    dialogConfig.width = "60%";
+    dialogConfig.data = {
+      // name: task.name,
+      // title: "TASK",
+      // description: task.description,
+      // actionButtonText: "Complete",
+      // productId: task.id
+      task: task,
+      actionButtonText: "TESTING"
```
Comment (bml2238) [inline frontend/crd/src/app/milestones-page/milestones/milestones.component.ts:53]: what is this testing button for? if we're not going to put in something here just remove it

File: frontend/crd/src/app/security/user.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export interface TempUser {
```
Comment (bml2238) [inline frontend/crd/src/app/security/user.ts:1]: did you add all the temp user stuff or is this from a different PR that hasn't been merged yet?

File: frontend/crd/src/app/tasks-modal/tasks-modal.component.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+<!--    <p style="font-size:30px"><b>Description: </b><br>-->
+<!--    {{ modalData.description }}</p>-->
+<!--    <p>Task Product ID: {{ modalData.productId}}</p>-->
+<!--  </section>-->
+  <div id="modal-body">
+    <div id="">
+      <p>{{modalData.task.yearLevel}}</p>
+      <p style="font-size:30px"><b>Description: </b><br>
+        {{ modalData.task.description }}</p>
+      <p>Task Product ID: {{ modalData.task.taskID}}</p>
```
Comment (bml2238) [inline frontend/crd/src/app/tasks-modal/tasks-modal.component.html:33]: remove this please the students don't need to see the task ID (why is it product?)

File: frontend/crd/src/app/tasks-modal/tasks-modal.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {TasksModalComponent} from './tasks-modal.component';
+import {Task, TaskType} from '../../domain/Task';
+import {YearLevel} from 'src/domain/Milestone';
+import {of} from "rxjs";
+import createSpyObj = jasmine.createSpyObj;
+
+// describe('TasksModalComponent', () => {
```
Comment (bml2238) [inline frontend/crd/src/app/tasks-modal/tasks-modal.component.spec.ts:8]: please put the tests back and add unit tests

File: frontend/crd/src/app/tasks-modal/tasks-modal.component.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  constructor(
+    public dialogRef: MatDialogRef<TasksModalComponent>,
+    @Inject(MAT_DIALOG_DATA) protected modalData: any,
+    public dialog: MatDialog,
+  ) {
+  }
+
+  ngOnInit() { }
+
+  actionFunction() {
```
Comment (bml2238) [inline frontend/crd/src/app/tasks-modal/tasks-modal.component.ts:20]: can you rename this to be more descriptive? is this just the close button?

File: frontend/crd/src/app/tasks/tasks.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {TasksComponent} from './tasks.component';
+import {of} from "rxjs";
+import {YearLevel} from "../../domain/Milestone";
+import {Task, TaskType} from "../../domain/Task";
+import {TasksService} from "./tasks.service";
+import createSpyObj = jasmine.createSpyObj;
+
+// describe('TasksComponent', () => {
```
Comment (bml2238) [inline frontend/crd/src/app/tasks/tasks.component.spec.ts:9]: put back the tests

File: frontend/crd/src/app/tasks/tasks.component.ts [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    public matDialog: MatDialog
+  ) {}
+
+  ngOnInit() {
+    this.tasksService.getTasks().subscribe((tasks: Task[]) => {
+      this.tasksList = tasks;
+    });
+  }
+
+  openLogoutModal() {
```
Comment (bml2238) [inline frontend/crd/src/app/tasks/tasks.component.ts:25]: remove this

File: frontend/crd/src/app/tasks/tasks.component.ts [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+      name: "logout",
+      title: "Are you sure you want to logout?",
+      description: "Pretend this is a convincing argument on why you shouldn't logout :)",
+      actionButtonText: "Logout",
+    }
+    // https://material.angular.io/components/dialog/overview
+    const modalDialog = this.matDialog.open(TasksModalComponent, dialogConfig);
+  }
+
+  openDeleteProductModal() {
```
Comment (bml2238) [inline frontend/crd/src/app/tasks/tasks.component.ts:42]: remove this

File: frontend/crd/src/app/tasks/tasks.service.spec.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    });
+    service = TestBed.inject(TasksService);
+    httpMock = TestBed.inject(HttpTestingController);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  // it('getMilestones should return list of milestones', () => {
```
Comment (bml2238) [inline frontend/crd/src/app/tasks/tasks.service.spec.ts:24]: put back the tests

File: frontend/crd/src/app/tasks/tasks.service.ts [hunk 13]
```diff
@@ -0,0 +1,47 @@
+import {Injectable} from '@angular/core';
```
Comment (bml2238) [inline frontend/crd/src/app/tasks/tasks.service.ts:1]: why are you using this task service and not the one in the util folder?

File: frontend/crd/src/domain/TaskList.ts [hunk 14]
```diff
@@ -0,0 +1,21 @@
+import {Task, TaskJSON} from "./Task";
```
Comment (bml2238) [inline frontend/crd/src/domain/TaskList.ts:1]: what is this object for? is this just the milestone?

File: frontend/crd/src/app/tasks/tasks.component.html [hunk 15]
```diff
@@ -0,0 +1,13 @@
+<div>
+  <mat-list class="matList">
+  <h3 mat-subheader class="matListTitle">Tasks</h3>
+<!--  <a mat-list-item class="matListItem" *ngFor="let task of tasksList" href={{task.id}}>-->
+<!--    {{task.name}} {{task.id}}-->
+<!--  </a>-->
+    <button mat-list-item class="matListItem" *ngFor="let task of tasksList" (click)="openTask(task)">{{task.name}} {{task.taskID}}</button>
+</mat-list>
+<!--  <main id="logout-button-holder">-->
```
Comment (bml2238) [inline frontend/crd/src/app/tasks/tasks.component.html:9]: please delete unused comments

**Review by bml2238 — CHANGES_REQUESTED** [review]
please put back all of the tests. if they aren't passing that indicates there is an issue with the code (or the test) that you should fix before pushing.

File: frontend/crd/src/app/dashboard/events/events.component.spec.ts [hunk 16]
```diff
@@ -1,7 +1,7 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { EventsComponent } from './events.component';
 import { MatCardModule } from "@angular/material/card";
-import { CarouselModule } from 'ngx-bootstrap/carousel';
+import { CarouselModule } from '@coreui/angular';
```
[the remark this answers is not in the document]
Comment (bml2238) [reply frontend/crd/src/app/dashboard/events/events.component.spec.ts]: this is still an issue please change it back

[the remark this answers is not in the document]
Comment (bml2238) [reply frontend/crd/src/app/dashboard/events/events.component.spec.ts]: was the carousel module not used or smth?

**Comment by bml2238** [comment]
what is the base-page component?

File: backend/src/main/resources/keys.json
Comment (BasketOfEggs) [inline backend/src/main/resources/keys.json:1]: This can be deleted

File: backend/src/main/java/com/senior/project/backend/security/domain/TempUser.java
Comment (BasketOfEggs) [inline backend/src/main/java/com/senior/project/backend/security/domain/TempUser.java:1]: This can be deleted

File: backend/src/main/java/com/senior/project/backend/util/URIBuilder.java
Comment (BasketOfEggs) [inline backend/src/main/java/com/senior/project/backend/util/URIBuilder.java:1]: This can be deleted

File: frontend/crd/src/app/security/user.ts (hunk 5 above)
Comment (BasketOfEggs) [reply frontend/crd/src/app/security/user.ts:1]: Yeah this stuff was mine. I'm pretty sure these are just merge artifacts

**Review by BasketOfEggs — CHANGES_REQUESTED** [review]

File: frontend/crd/src/app/tasks-modal/tasks-modal.component.html [hunk 17]
```diff
@@ -0,0 +1,20 @@
+<div id="modal-content-wrapper">
+  <header id="modal-header">
+    <h1 id="modal-title"> {{ modalData.task.name }}</h1>
+  </header>
+  <div id="modal-body">
+    <div id="">
+      <p>{{modalData.task.yearLevel}}</p>
```
Comment (bml2238) [inline frontend/crd/src/app/tasks-modal/tasks-modal.component.html:7]: please remove this

**Review by bml2238 — CHANGES_REQUESTED** [review]
please implement any requested changes (deleting the base-page component and other files) and fix the merge conflicts and then this is good to go

**Review by BasketOfEggs — APPROVED** [review]
Once you fix the merge conflicts Im good with this
